at-quannguyen/W4/main.rb

```diff
@@ -0,0 +1,621 @@
+    clear
+    puts '========= Delete Notification ========='
+    print 'notification id : '
+    notification_id = begin
+                        Integer(gets.chomp)
+                      rescue StandardError
+                        false
+                      end
+    if notification_id == false
+      puts 'input id not string !'
+      csleep
+      cexit
+      action_delete_notification(team_id)
+    end
+
+    if @notification_controller.id_by_notification(notification_id).count.zero?
+      puts 'notification id not find !'
+      csleep
+      cexit
+      action_delete_notification(team_id)
+    end
+
+    if @notification_controller.detele(team_id, notification_id)
+      puts 'delete notification complete !'
+      csleep
+      action_notification_manager
+    else
+      puts 'delete notification fail !'
+      csleep
+      cexit
+      action_delete_notification(team_id)
+    end
+  end
+
+  def action_csv_manager
+    clear
+    puts '========= CSV Manager ========='
+    puts '1. Import file'
+    puts '2. Export file'
+    print 'Action : '
+    number = begin
+               Integer(gets.chomp)
+             rescue StandardError
+               false
+             end
+
+    if number == false
+      puts 'input not string !'
+      csleep
+      cexit
+      action_csv_manager
+    end
+
+    action_import_csv if number == 1
+    action_export_csv if number == 2
+  end
+
+  def action_import_csv
+    path = 'employee.csv'
+    @csv_controller.register_from_csv(path)
+  end
+
+  def action_export_csv
+    @csv_controller.output_users_to_csv
+  end
+
+  def action_leave_manager
+    clear
+    account = @session_controller.current_user
+    query = account.to_a[0]
+    team_id = query['team_id']
+    role = query['role']
+    puts '========= Leave Manager ========='
+    puts '1. Send a leave request'
+    puts '2. Show list leaves request need approve' if role == 2
+    puts '5. Approve or reject leave request' if role == 2
+    puts '6. Update a leave request'
+    puts '7. Show list of leaves of employees'
+    puts '3. Back'
+    puts '4. Logout'
+    puts '5. Exit program'
+    print 'Action : '
+    number = begin
+               Integer(gets.chomp)
+             rescue StandardError
+               false
+             end
+
+    if number == false
+      puts 'input not string !'
+      csleep
+      cexit
+      action_leave_manager
+    end
+    action_send_leave(team_id) if number == 1
+  end
+
+  def action_send_leave(team_id)
+    puts '========= Leave Manager ========='
+    print 'reason : '
+    reason = gets.chomp
+    puts ''
+    print 'total date off : '
+    total_date_off = begin
+                       Integer(gets.chomp)
+                     rescue StandardError
+                       false
+                     end
+
+    leave = Leave.new.tap do |e|
+      e.team_id = team_id
+      e.reason = reason
+      e.total_date_off = total_date_off
+      e.status = 0
+      e.publish_date = Time.new
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/7deb92b5e1c7275222ca44a289eebcc27cbc38fe/at-quannguyen/W4/main.rb#L574"></span>Rails/TimeZone: Do not use <code>Time.new</code> without zone. Use one of <code>Time.zone.now</code>, <code>Time.current</code>, <code>Time.new.in_time_zone</code>, <code>Time.new.utc</code>, <code>Time.new.getlocal</code>, <code>Time.new.xmlschema</code>, <code>Time.new.iso8601</code>, <code>Time.new.jisx0301</code>, <code>Time.new.rfc3339</code>, <code>Time.new.httpdate</code>, <code>Time.new.to_i</code>, <code>Time.new.to_f</code> instead. (https://github.com/rubocop-hq/rails-style-guide#time, http://danilenko.org/2012/7/6/rails_timezones)</td>
    </tr>
  </tbody>
</table>